assets/product-images-slider.js

```diff
@@ -0,0 +1,91 @@
+const numberOfThumbnails = document.querySelector('[data-glide-thumbnail-slides]').dataset.glideThumbnailSlides;
+
+const glideOptions = {
+  image: {
+    type: 'carousel',
+    startAt: 0,
+    perView: 1,
+    focusAt: 'center',
+    gap: 0,
+    dragThreshold: 10,
+    peek: 75,
+    breakpoints: {
+      1200: {
+        peek: 70
+      },
+      1000: {
+        peek: 55
+      },
+      875: {
+        peek: 45
+      },
+      749: {
+        peek: 65
+      },
+      550: {
+        peek: 55
+      },
+      375: {
+        peek: 45
+      },
+    }
+  },
+  thumbnail: {
+    type: 'carousel',
+    startAt: 0,
+    perView: numberOfThumbnails,
+    focusAt: numberOfThumbnails % 2 === 0 ? (numberOfThumbnails / 2) - 1 : 'center'
+  }
+};
+
+const createGlideInstance = (element, optionsObject) => {
+  return new Glide(element, optionsObject);
+};
+
+const mountGlideInstance = (instance) => {
+  instance.mount()
+};
+
+const indexOfSlide = (instance, identifier) => {
+  instance.on('mount.after', function() {
+    instance.on('run', function() {
+      console.log(`${identifier} index: ${instance.index}`)
+    })
+  })
+};
+
+const onSlide = (instance, staticInstance) => {
```

Comment:
Be aware that currently, you do not have any sort of scoping setup for your javascript, so your javascript variables are in the global scope.
A variable name like onSlide is very dangerous in the global scope.

Come up with some sort scheme to keep your variables in specific scope, while still being able to take advantage of having your js in seperate files.